src/GafferCycles/SocketHandler.cpp

```diff
@@ -514,27 +514,29 @@ void setupLightPlugs( const std::string &shaderName, const ccl::NodeType *nodeTy
 
 	std::set<const Plug *> validPlugs;
 
-	if( shaderName == "portal" )
-	{
-		validPlugs.insert( setupTypedPlug<FloatPlug>( "size", plugsParent, Gaffer::Plug::In, 2.0f ) );
-	}
-	else
+	if( shaderName != "portal" )
 	{
 		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "cast_shadow" ) )), plugsParent, Gaffer::Plug::In ) );
-		//validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "use_mis" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupTypedPlug<BoolPlug>( "use_mis", plugsParent, Gaffer::Plug::In, true ) );
+		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "use_camera" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "use_diffuse" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "use_glossy" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "use_transmission" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "use_scatter" ) )), plugsParent, Gaffer::Plug::In ) );
+		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "use_caustics" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "max_bounces" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupPlug( nodeType, *(nodeType->find_input( ccl::ustring( "lightgroup" ) )), plugsParent, Gaffer::Plug::In ) );
 		validPlugs.insert( setupTypedPlug<FloatPlug>( "intensity", plugsParent, Gaffer::Plug::In, 1.0f ) );
 		validPlugs.insert( setupTypedPlug<FloatPlug>( "exposure", plugsParent, Gaffer::Plug::In, 0.0f ) );
 		validPlugs.insert( setupTypedPlug<Color3fPlug>( "color", plugsParent, Gaffer::Plug::In, Color3f( 1.0f ) ) );
 	}
 
-	if( shaderName == "spot_light" )
+	if( shaderName == "portal" || shaderName == "quad_light" || shaderName == "disk_light" )
+	{
+		validPlugs.insert( setupTypedPlug<FloatPlug>( "sizeu", plugsParent, Gaffer::Plug::In, 2.0f ) );
+		validPlugs.insert( setupTypedPlug<FloatPlug>( "sizev", plugsParent, Gaffer::Plug::In, 2.0f ) );
```

Comment:
What do you think about representing these as `width` and `height` rather than `sizeu` and `sizev`? It's true we often present a renderer's parameters mostly raw and unvarnished to the user, but `width` and `height` are likely more familiar/expected names as well as being closer to UsdLuxLight.

We'll likely add the same to Arnold quad lights at some point in the near future too.

Reply:
I'm inclined to agree with Murray here - what do you think @boberfly? Are you likely to be able to find time to get this over the line, or would you like us to make those final few tweaks?

Reply:
Hey @murraystevenson & @johnhaddon apologies this wasn't addressed before the release, but I got this sorted now with width/height! This reads better than sizeu/sizev.